source/state_representation/include/state_representation/space/cartesian/CartesianState.hpp

```diff
@@ -598,4 +599,87 @@ inline void swap(CartesianState& state1, CartesianState& state2) {
   std::swap(state1.force_, state2.force_);
   std::swap(state1.torque_, state2.torque_);
 }
+
+/**
+ * @brief Convert a string to a CartesianStateVariable enum (case insensitive)
+ * @throws exceptions::InvalidStateException
+ * @param variable The string to convert
+ * @return A CartesianStateVariable enum corresponding to the input string
+ */
+inline state_representation::CartesianStateVariable string_to_cartesian_state_variable(const std::string& variable) {
+  std::string case_insensitive_variable;
+  std::transform(variable.begin(), variable.end(), std::back_inserter(case_insensitive_variable), [](unsigned char c) {
+    return std::tolower(c);
+  });
+  if (case_insensitive_variable == "position") {
+    return CartesianStateVariable::POSITION;
+  } else if (case_insensitive_variable == "orientation") {
+    return CartesianStateVariable::ORIENTATION;
+  } else if (case_insensitive_variable == "pose") {
+    return CartesianStateVariable::POSE;
+  } else if (case_insensitive_variable == "linear_velocity") {
+    return CartesianStateVariable::LINEAR_VELOCITY;
+  } else if (case_insensitive_variable == "angular_velocity") {
+    return CartesianStateVariable::ANGULAR_VELOCITY;
+  } else if (case_insensitive_variable == "twist") {
+    return CartesianStateVariable::TWIST;
+  } else if (case_insensitive_variable == "linear_acceleration") {
+    return CartesianStateVariable::LINEAR_ACCELERATION;
+  } else if (case_insensitive_variable == "angular_acceleration") {
+    return CartesianStateVariable::ANGULAR_ACCELERATION;
+  } else if (case_insensitive_variable == "acceleration") {
+    return CartesianStateVariable::ACCELERATION;
+  } else if (case_insensitive_variable == "force") {
+    return CartesianStateVariable::FORCE;
+  } else if (case_insensitive_variable == "torque") {
+    return CartesianStateVariable::TORQUE;
+  } else if (case_insensitive_variable == "twist") {
+    return CartesianStateVariable::TWIST;
+  } else if (case_insensitive_variable == "wrench") {
+    return CartesianStateVariable::WRENCH;
+  } else if (case_insensitive_variable == "all") {
+    return CartesianStateVariable::ALL;
+  } else {
+    throw exceptions::InvalidStateException("Invalid Cartesian state variable: " + variable);
```

Comment:
I would prefer the more verbose option `InvalidStateVariableException` here, the `InvalidStateException` refers too much to the state itself IMO

Reply:
yes, fair enough, thanks